src/common/expression/PropertyExpression.cpp

```diff
@@ -68,15 +68,29 @@ void TagPropertyExpression::accept(ExprVisitor* visitor) {
 }
 
 const Value& InputPropertyExpression::eval(ExpressionContext& ctx) {
-  return ctx.getInputProp(prop_);
+  if (!propIndex_.has_value()) {
+    auto indexResult = ctx.getInputPropIndex(prop_);
+    if (!indexResult.ok()) {
+      return Value::kEmpty;
+    }
+    propIndex_ = indexResult.value();
+  }
+  return ctx.getColumn(propIndex_.value());
 }
```

Comment:
I remember we have a `ColumnIndexExpr`?  It looks like that rewriteing `InputPropExpr` to `ColumnIndexExpr` is more easy

Reply:
Good job, we can add an optimize rule to do it. But current optimization is also useful now and no need to rewrite expression.